src/main/java/net/coderbot/iris/uniforms/PropertiesUniforms.java

```diff
@@ -0,0 +1,82 @@
+package net.coderbot.iris.uniforms;
+
+import net.coderbot.iris.Iris;
+import net.coderbot.iris.gl.program.ProgramBuilder;
+import net.coderbot.iris.gl.uniform.UniformUpdateFrequency;
+import net.coderbot.iris.shaderpack.IdMapParser;
+import net.minecraft.block.Block;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.render.block.entity.BlockEntityRenderDispatcher;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.Hand;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import java.util.Map;
+import java.util.function.IntSupplier;
+
+public final class PropertiesUniforms {
+
+    private PropertiesUniforms(){}
+
+    public static void addPropertiesUniforms(ProgramBuilder builder) {
+        builder
+           .uniform1i(UniformUpdateFrequency.PER_FRAME, "heldItemId", getHeldItemId(Hand.MAIN_HAND))
+           .uniform1i(UniformUpdateFrequency.PER_FRAME, "heldItemId2", getHeldItemId(Hand.OFF_HAND))
+           .uniform1i(UniformUpdateFrequency.PER_FRAME, "blockEntityId", PropertiesUniforms::getBlockEntityId)
+           .uniform1i(UniformUpdateFrequency.PER_FRAME, "entityId", PropertiesUniforms::getEntityId);
+
+    }
+
+    /**
+     * A method that returns the heldItemId of the item the player is currently holding
+     * @param hand offhand or main hand (used to not duplicate code)
+     * @return int supplier of the id of the item
+     */
+    private static IntSupplier getHeldItemId(Hand hand) {
+        if (MinecraftClient.getInstance().player != null) {
+            Identifier currentStack = Registry.ITEM.getId(MinecraftClient.getInstance().player.getStackInHand(hand).getItem());
+            IdMapParser parser = Iris.getPipeline().getPack().getIdMapParser();
+            if (parser.getItemProperties().containsKey(currentStack)) {
+                return () -> parser.getItemProperties().get(currentStack);
+            }
+        }
+        return () -> -1;
+    }
```

Comment:
This isn't how uniform attachments work. `getHeldItemId` will be called exactly one time for a given shader pack when the shader pipeline is compiled, and then never again. This will cause subtle bugs depending on what item the player is holding at the time that the shader pipeline is initialized.